mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulCreateTournamentScreen.kt

```diff
@@ -0,0 +1,199 @@
+package ch.epfl.sdp.mobile.state
+
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.core.text.isDigitsOnly
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.tournaments.TournamentFacade
+import ch.epfl.sdp.mobile.application.tournaments.TournamentReference
+import ch.epfl.sdp.mobile.ui.i18n.LocalizedStrings
+import ch.epfl.sdp.mobile.ui.tournaments.CreateDialog
+import ch.epfl.sdp.mobile.ui.tournaments.CreateDialogState
+import ch.epfl.sdp.mobile.ui.tournaments.CreateDialogState.*
+import kotlin.math.log2
+import kotlin.math.pow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * A composable that makes a [CreateDialog] stateful.
+ *
+ * @param user the current [AuthenticatedUser].
+ * @param navigateToTournament The action to navigate to a certain tournament via its reference.
+ * @param cancelClick The action to take when clicking on the cancel button.
+ * @param modifier The [Composable]'s modifier.
+ */
+@Composable
+fun StatefulCreateTournamentScreen(
+    user: AuthenticatedUser,
+    navigateToTournament: (reference: TournamentReference) -> Unit,
+    cancelClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  val scope = rememberCoroutineScope()
+
+  val tournamentFacade = LocalTournamentFacade.current
+  val strings = LocalLocalizedStrings.current
+
+  val actions =
+      rememberUpdatedState(
+          CreateTournamentDialogActions(
+              navigateToTournament = navigateToTournament,
+              cancelClick = cancelClick,
+          ),
+      )
+
+  val state =
+      remember(user, actions, scope, tournamentFacade) {
```

Comment:
Missing `strings`

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/336/commits/aa97a4ab2979585c281119d700462b95b1fe9884